### PR TITLE
[cdac] link a stub contract descriptor if cdac-build-tool is not available

### PR DESCRIPTION
**Pull request by lambdageek**
Fixes builds that use src/coreclr/build-runtime.sh directly, or bringup scenarios without msbuild

Use `DOTNET_HOST_PATH` to run `cdac-build-tool` if it is set.  Fixes VMR source builds: https://github.com/dotnet/installer/pull/19534#issuecomment-2066570507

Related to https://github.com/dotnet/runtime/pull/100650#discussion_r1572203146

### CODE REVIEW

File: src/coreclr/runtime.proj [hunk 1]
```diff
@@ -50,6 +50,8 @@
       <_CoreClrBuildArg Condition="'$(HostCrossOS)' != ''" Include="-hostos $(HostCrossOS)" />
       <_CoreClrBuildArg Include="-outputrid $(OutputRID)" />
       <_CoreClrBuildArg Condition="'$(BuildSubdirectory)' != ''" Include="-subdir $(BuildSubdirectory)" />
+      <_CoreClrBuildArg Include="-cmakeargs &quot;-DCLR_DOTNET_HOST_PATH=$(DOTNET_HOST_PATH)&quot;" Condition="'$(DOTNET_HOST_PATH)' != ''" />
+      <_CoreClrBuildArg Include="-cmakeargs -DCLR_DOTNET_HOST_PATH=dotnet" Condition="'$(DOTNET_HOST_PATH)' == ''" />
```
Comment (am11) [inline src/coreclr/runtime.proj:54]: I think only .NET Framework doesn't have DOTNET_HOST_PATH set, which is not supported by runtime.proj. Did you notice build failing somewhere without this line?

Comment (lambdageek) [reply src/coreclr/runtime.proj:54]: No, I saw `eng/illink.targets`, but I realize now it's talking about .NET Framework. I'll remove this.

**Review by am11 — APPROVED** [review]
LGTM, thanks!

**Review by elinor-fung — APPROVED** [review]

**Comment by AndyAyersMS** [comment]
FYI the issue this is fixing is causing all codegen PR formatting legs to fail...

**Comment by lambdageek** [comment]
> all codegen PR formatting legs

Not sure which ones you mean.  Can you show me an example so I know what to watch for in the future

**Comment by am11** [comment]
CI is green on AzDO. Feel free to merge.

**Pull request merged**